File: packages/typescript/src/network/batches-api.ts
import { DedHttpClient } from './http-client';
import type { DataResponse, BatchDetail, FingerprintDetail } from './types';

/**
 * API client for batch endpoints (all public, no API key required).
 */
export class BatchesApi {
  constructor(private readonly http: DedHttpClient) {}

  /**
   * Get batch details by ID.
   *
   * @param batchId - UUID of the batch
   * @returns Batch detail including status and fingerprint count
   */
  async get(batchId: string): Promise<DataResponse<BatchDetail>> {
    return this.http.getPublic<DataResponse<BatchDetail>>(`/v1/batches/${batchId}`);
  }

  /**
   * Get all fingerprints in a batch.
   *
   * @param batchId - UUID of the batch
   * @returns List of fingerprint details belonging to this batch
   */
  async getFingerprints(batchId: string): Promise<DataResponse<FingerprintDetail[]>> {
    return this.http.getPublic<DataResponse<FingerprintDetail[]>>(
      `/v1/batches/${batchId}/fingerprints`
    );
  }
}

File: packages/python/src/constellation_digital_evidence_sdk/network/batches_api.py
"""API client for batch endpoints (all public, no API key required)."""

from __future__ import annotations

from typing import Any

from .http_client import DedHttpClient


class BatchesApi:
    """API client for batch endpoints (all public)."""

    def __init__(self, http: DedHttpClient) -> None:
        self._http = http

    async def get(self, batch_id: str) -> dict[str, Any]:
        """Get batch details by ID (public).

        Args:
            batch_id: UUID of the batch.
        """
        return await self._http.get_public(f"/v1/batches/{batch_id}")

    async def get_fingerprints(self, batch_id: str) -> dict[str, Any]:
        """Get all fingerprints in a batch (public).

        Args:
            batch_id: UUID of the batch.
        """
        return await self._http.get_public(f"/v1/batches/{batch_id}/fingerprints")
